### LCM
#### LCM looks for closed itemset with respect to an input minimum support

In [2]:
from skmine.datasets.fimi import fetch_chess
chess = fetch_chess()
chess.head()

0    [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25...
1    [1, 3, 5, 7, 9, 12, 13, 15, 17, 19, 21, 23, 25...
2    [1, 3, 5, 7, 9, 12, 13, 16, 17, 19, 21, 23, 25...
3    [1, 3, 5, 7, 9, 11, 13, 15, 17, 20, 21, 23, 25...
4    [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25...
Name: chess, dtype: object

In [3]:
chess.shape

(3196,)

In [4]:
from skmine.itemsets import LCM
lcm = LCM(min_supp=2000, n_jobs=4)
# minimum support of 2000, running on 4 processes
%time patterns = lcm.fit_transform(chess)

CPU times: user 174 ms, sys: 33.1 ms, total: 207 ms
Wall time: 12 s


In [5]:
patterns.shape

(68967, 2)

In [6]:
patterns[patterns.itemset.map(len) > 3]
# simple syntax for post hoc filtering

itemset support
14         (25, 58, 17, 29)    2179
17          (3, 58, 17, 29)    2272
18          (25, 58, 3, 29)    2520
22         (25, 29, 58, 31)    2220
23          (29, 58, 3, 31)    2360
...                     ...     ...
68958       (52, 54, 5, 40)    2034
68959   (29, 54, 5, 40, 52)    2027
68964      (70, 52, 40, 29)    2001
68965      (70, 58, 40, 29)    2006
68966  (29, 70, 58, 40, 52)    2000

[67716 rows x 2 columns]

------------
`Note`

Even when setting a very high minimum support threshold, we discovered more than 60K from only 3196 original transactions.
This is a good illustration of the so-called **pattern explosion problem**